File: oscexchange/src/main/java/ac/at/fhstp/digitech/oscexchange/errors/OSCErrorListener.java
package ac.at.fhstp.digitech.oscexchange.errors;

/**
 * A listener for when an error occurs
 */
@FunctionalInterface
public interface OSCErrorListener {

    /**
     * Handles the occurred error
     *
     * @param error The error that occurred
     */
    void handle(OSCError error);

}

File: oscexchange/src/main/java/ac/at/fhstp/digitech/oscexchange/errors/OSCErrorListener.kt
package ac.at.fhstp.digitech.oscexchange.errors

/**
 * A listener for when an error occurs
 */
fun interface OSCErrorListener {
    /**
     * Handles the occurred error
     *
     * @param error The error that occurred
     */
    fun handle(error: OSCError)
}